src/cljs/main/broadfcui/page/workspace/summary/catalog/wizard.cljs

```diff
@@ -118,27 +166,53 @@
      (when (not= (:page-num prev-state) (:page-num @state))
        (react/call :scroll-to (@refs "scroller") 0)))
    :next-page
-   (fn [{:keys [props state refs this locals after-update]}]
+   (fn [{:keys [state refs this locals after-update props]}]
      (swap! state dissoc :validation-error)
      (if-let [error-message (react/call :validate (@refs "wizard-page"))]
        (swap! state assoc :validation-error error-message)
-       (let [attributes-from-page (react/call :get-attributes (@refs "wizard-page"))]
-         (swap! state update :working-attributes merge attributes-from-page)
-         (swap! locals update :page-attributes assoc (:page-num @state) attributes-from-page)
-         (after-update
-          #(if (< (:page-num @state) (-> props :library-schema :wizard count dec))
-             (swap! state update :page-num inc)
-             (react/call :submit this))))))
+       (let [{:keys [working-attributes pages-seen required-attributes page-num]} @state
+             attributes-from-page (react/call :get-attributes (@refs "wizard-page"))
+             all-attributes (merge working-attributes attributes-from-page)
+             invalid-attributes (atom #{})]
+         (swap! state assoc :working-attributes all-attributes)
+         (swap! locals update :page-attributes assoc page-num attributes-from-page)
+         (doseq [page (conj pages-seen page-num)]
+              (let [questions (first (get-questions-for-page all-attributes (:library-schema props) page))
```

Comment:
Instead of using `first`, I would suggest destructuring and throwing away the unwanted item:
```clojure
(let [[questions _] (get-questions...
```